### DISCUSSION TITLE
Prepared statements

### DISCUSSION THREAD

**Discussion by JulienZolli** [ibis-project/ibis#9262] [Q&A] [answered]
Hi,

I'm currently using Ibis in an API where users can execute some pre-defined Ibis queries with different parameters.
When profiling my API, I saw that Ibis is taking ~40ms to compile my query which is half the response time of my API !

I replaced Ibis with hard-coded prepared statements but I would still like to benefits from the flexibility of Ibis : Is there a way to output prepared statements with Ibis ?
This way I could create my queries with Ibis and store the prepared statement for later use !

**Comment by cpcloud** [comment]
Hi, thanks for opening a discussion!

Ibis doesn't support outputting prepared statements. It's also not currently on the roadmap.

Can you give some pseudocode showing what such a feature would look like?

**Comment by cpcloud** [comment, marked as answer]
There have been some recent performance enhancements for expression construction in certain common expressions like `relocate`, `rename`, `drop` and some improvements to the `c` selector's performance as well.

Closing as resolved!
